### ISSUE TITLE
DRP fails to update status when next song has same title

### ISSUE THREAD

**Issue by LyingJay** [jojo2357/Music-Discord-Rich-Presence#32]
Basically, some of my songs have the same title in the metadata, since it is the same song. Because of this, they also have the same album art, and album name (the title of the song is the album name, like a single album). However, I may have variants of the song, such as abridged/full versions, and different versions by different singers, or covers. When Groove moves from one song to the next in the playlist, the change fails to be reflected in DRP, and the artist remains the same.

e.g. I have a song "SONG", and I have two versions, one sung by Singer A and the other by Singer B. Both versions have the same album art. Both songs also have the same metadata, excluding the artist. When I play SONG by Singer A, DRP displays Title: SONG and Artist: Singer A. When the song finishes and moves to SONG by Singer B, DRP still displays Title: SONG and Artist: Singer A.

This also occurs if I click the skip button or start playing a variant of the song, that is, Groove does not move to the song itself and I manually play it.
The issue does not occur if I play a different song and then play the variant. It only occurs when the 'same' song is played in succession.
I'm not sure if my explanation above is clear, as to me it feels a bit wordy, so feel free to ask me for clarification/further details.

Also, I'm glad to let you know that my previous issue where the RunHidden shortcut had a weird error no longer occurs in the latest release, so good job there!

**Comment by jojo2357** [comment]
No I know exactly why this is happening. Sort of a relic from a couple of versions back I didn't bother to fix the issue since nobody had it and I wasn't able to produce it myself but I know exactly the cause and fix. It stems from trying to save as many resources as possible so I guess we gonna be using a bit more resources.

Glad to hear random issues are getting random fixes xD and glad to see you're still around

Did you see the library repo? Its currently broken atm but theres some really useful stuff there

**Comment by LyingJay** [comment]
Haven't seen the library repo yet, and I don't fully know what it's about but I assume it's a big library of album art, made by the users, so that users (eventually) no longer need .dat files and discord applications because the library repo will find the album art for your music. If so, I'd definitely be interested in contributing (although I really don't know if anyone else that uses DRP listens to the music I listen to, haha).

I assume that by trying to save resources, it was so that DRP wouldn't always check in the background for when a new song started.

It's very encouraging to see you're still working hard on this project and expanding it as well.

**Comment by jojo2357** [comment]
The point is also for you to extract album art from mp3's really easily and then upload them. You would have to go into a previous commit to use it or just keep an eye on it for now.

I find it so funny that the original repo gets [issues about album art](https://github.com/dsdude123/GroovyRP/issues/23) every once in a while and they keep getting wontfix labels yet here I am with that support already :/

**Comment by jojo2357** [comment]
@LyingJay please verify most recent commit

**Comment by LyingJay** [comment]
yep. it works now!

**Comment by jojo2357** [comment]
See most recent release
